doc/ChangeLog

```diff
@@ -1,4 +1,81 @@
 ===============================================================
+Tag name:  ctsm5.1.dev053
+Originator(s):  slevis (Samuel Levis,SLevis Consulting,303-665-1310)
+Date:  Tue Aug 10 16:22:24 MDT 2021
+One-line Summary: Adding history variables for use with Newton-Krylov
+
+Purpose and description of changes
+----------------------------------
+
+ Considering @klindsay28's Newton-Krylov spin-up method for BGC and MIMICS.
+ Newton-Krylov needs decomp_k and pathfrac_decomp_cascade on ctsm history
+ files.
+
+ decomp_k was already in place.
+ I have now also added pathfrac_decomp_cascade.
+
+Significant changes to scientifically-supported configurations
+--------------------------------------------------------------
+
+Does this tag change answers significantly for any of the following physics configurations?
+(Details of any changes will be given in the "Answer changes" section below.)
+
+    [Put an [X] in the box for any configuration with significant answer changes.]
+
+[ ] clm5_1
+
+[ ] clm5_0
+
+[ ] ctsm5_0-nwp
+
+[ ] clm4_5
+
+
+Bugs fixed or introduced
+------------------------
+Issues fixed (include CTSM Issue #): #1455
+
+Known bugs found since the previous tag (include issue #):
+ I did not open an issue for this: decomp_k was dimensioned wrong, and I
+ corrected it in this PR without affecting answers. In particular I changed the
+ size of the third dimension from ndecomp_pools to ndecomp_cascade_transitions.
+
+Notes of particular relevance for users
+---------------------------------------
+Substantial timing or memory changes:
+[For timing changes, you should at least check the PFS test in the test
+
```

Comment:
are these history fields going to be written out for each column or grid cell average?  I'd assume the former, but wondered if N-K solver can handle column level history files @klindsay28?

@ekluzek I also wondered how we easily turn on this output for a N-K spinup?  Should this feature come into this PR, or just handled more manually at this stage?

Reply:
Regarding the second question:
One can activate so-called "inactive" history fields using the namelist. In this testing phase I would rely on that. If after the testing we decide to go with Newton-Krylov, then we can create a corresponding compset. Would you respond differently, @ekluzek ?

Reply:
> are these history fields going to be written out for each column or grid cell average? I'd assume the former, but wondered if N-K solver can handle column level history files @klindsay28?

When added to the primary history file, history fields appear as grid cell average by default. With appropriate namelist selections, they can also be sent to history as column variables.

Reply:
@klindsay28, grid cell average variables are dimensioned by lat, lon, while column variables are written as a column vector. The latter complicates a bit the task of finding where you are on the planet, which I think is why @wwieder asked the question.

Reply:
The solver will be reading and modifying restart files. Are these column based? (I don't have the CLM/CTSM diagram of grid cell/PFT/column handy.) If so, then the solver will be dealing with column level variables, and column level history output then seems natural. If there is history output needed for the preconditioner that is cell averages, then I think there will be an extra step of mapping the column index to the proper grid cell.